Assets/Scripts/Models/Buildable/Furniture.cs

```diff
@@ -1306,6 +1306,20 @@ public void SetRotation(float rotation)
         Rotation = rotation;
     }
 
+    public Tile[] GetAllTiles()
+    {
+        Tile[] tiles = new Tile[Height * Width];
+        for (int x = 0; x < Width; x++)
+        {
+            for (int y = 0; y < Height; y++)
+            {
+                tiles[x + (y * Width)] = World.Current.GetTileAt(Tile.X + x, Tile.Y + y, Tile.Z);
+            }
+        }
+
+        return tiles;
+    }
+
```

Comment:
As far as I understand it, using an Ienumerable leads to more object creation, of course my use of linq in PowerConnection may make that a moot point.

And at least how I'm using it (only when a furniture is placed or a utility is placed in a furniture's tiles) it will be such a small difference either way, I think it will be virtually impossible to see a difference either way.

I do recall reading that arrays in foreach loops don't have the excess object creation issues that other objects do in unity's mono, and also that that is how some of the linq statements are implemented, so that *should* be some savings there.

Either way, I think for now either one would be near enough to identical due to the small usage, and if the function starts seeing greater use, it can be revisited to see if one or the other has a more significant difference.